### PR TITLE
Add Cosmos3 Edge UniPC scheduler

### PR DESCRIPTION
**Pull request by atharvajoshi10**
# What does this PR do?
- Bug Fix for Flow Scheduler
- Fix metadata params

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_cosmos3_unipc.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+import numpy as np
+import torch
+
+from .scheduling_unipc_multistep import UniPCMultistepScheduler
+
+
+class Cosmos3EdgeUniPCMultistepScheduler(UniPCMultistepScheduler):
+    """UniPC scheduler with the flow schedule and arithmetic used by Cosmos3 Edge."""
+
+    def set_cosmos3_edge_native_flow_timesteps(
```
Comment (yiyixuxu) [inline src/diffusers/schedulers/scheduling_cosmos3_unipc.py:24]: can you remove this scheduler and just change the config for scheduler in https://huggingface.co/nvidia/Cosmos3-Edge/blob/main/scheduler/scheduler_config.json

Comment (atharvajoshi10) [reply src/diffusers/schedulers/scheduling_cosmos3_unipc.py:24]: Thanks for the feedback regarding removal of the Scheduler

I tried the config-only version locally.

```
  {
    "_class_name": "UniPCMultistepScheduler",
    "use_flow_sigmas": true,
    "use_karras_sigmas": false,
    "flow_shift": 3.0
  }

```
However, that config alone does not fully reproduce the a similar quality Cosmos Framework result.

Edge uses a non-default base flow grid: it starts at` 1 - 1 / num_train_timesteps` and runs toward zero, whereas stock UniPC’s default flow grid starts at `1.0` and runs toward `1 / num_train_timesteps`

Is there any other way to define custom scheduler behavior?
